firmware/src/core1_sensor.cpp

```diff
@@ -0,0 +1,24 @@
+#include <core1_sensor.h>
+
+Adafruit_BME680 bme688(spi0,BME688_PICO_PIN,BME688_POCI_PIN,BME688_SCK_PIN,BME688_CS_PIN,true);
+
+bool core1_setup() {
+  bool rslt = bme688.begin(0,true);
+
+  queue_init(&sensor_queue,sizeof(sensor_data_t),32);
+
+  return rslt;
+}
+
+void core1_main()
+{
+  while(true) {
+    bme688.performReading(); // reading takes ~ 400 ms
+    
+    sensor_data_t data;
+    data.pressure_pa = bme688.pressure;
+    data.temperature_c = bme688.temperature;
+    data.humidity_units = bme688.humidity;
+    queue_add_blocking(&sensor_queue,&data);
```

Comment:
This is effectively calling [`memcpy`](https://github.com/raspberrypi/pico-sdk/blob/efe2103f9b28458a1615ff096054479743ade236/src/common/pico_util/queue.c#L48) under the hood, so we will need to make sure the underlying structs are both byte aligned.

That's true for `app_regs_t` but not `sensor_data_t`, so you'll want to wrap the other struct in a `#pragma` packing directive that matches the other one.